python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -462,20 +468,26 @@ def all_nodes_set(self) -> Union[ItemSet, ItemSetDict]:
         self._check_loaded()
         return self._all_nodes_set
 
-    def _init_tasks(self, tasks: List[OnDiskTaskData]) -> List[OnDiskTask]:
+    def _init_tasks(
+        self, tasks: List[OnDiskTaskData], selected_task: list
+    ) -> List[OnDiskTask]:
         """Initialize the tasks."""
         ret = []
         if tasks is None:
             return ret
         for task in tasks:
-            ret.append(
-                OnDiskTask(
-                    task.extra_fields,
-                    self._init_tvt_set(task.train_set),
-                    self._init_tvt_set(task.validation_set),
-                    self._init_tvt_set(task.test_set),
+            if (
+                selected_task == ["all"]
+                or task.extra_fields["name"] in selected_task
```

Comment:
as `name` is not a required field for now, accessing it blindly may throw exception here.